cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -827,6 +830,14 @@ installBuiltExe verbosity overwritePolicy
       then removeDirectory destination
       else removeFile      destination
     copy = copyFile source destination >> pure True
+    overwrite :: IO Bool
+    overwrite = remove >> copy
+    maybeOverwrite :: IO Bool
+    maybeOverwrite = do
+      a <- promptYesNo
+        "Existing file found while installing executable. Do you want to unlink that file? (y/n)"
+        MandatoryPrompt
+      if a then overwrite else pure a
```

Comment:
You're doing this work twice, here and in `InstallSymlink.hs`. Could this maybe be unified? :)

Reply:
I think it definitely might be able to do some refacturing to increase sharing between `installBuiltExe` and `symlinkBinary`. I'm not sure it's warranted in the case of `maybeOverwrite`. I'm personally not convinced that the code is made more legible by just doing it for `maybeOverwrite`, but I've included it in the latest commit (9f594be790e6cc6fe19485c1d7c655bd09ecf07b) for you to be the judge :)